src/app/shared/pipes/search-by/search-by.pipe.ts

```diff
@@ -0,0 +1,20 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({
+  name: 'searchBy',
+})
+export class SearchByPipe implements PipeTransform {
+  transform(array: any[], field: string, searchString: string): any[] {
+    if (!Array.isArray(array)) {
+      return [];
+    }
+
+    const filteredArray = array.filter(item => {
+      return item[field].includes(searchString)
+        || item[field].toLowerCase().includes(searchString)
+        || item[field].toUpperCase().includes(searchString);
+    });
```

Comment:
Please mutate both arguments (item[field] and searchString) to single case. It will gave little performance boost to the filter method and provide this function with ability to handle cpitalized words such as 'Lorem'

Reply:
Yes, good catch. You are right!